FirebaseCore.podspec

```diff
@@ -32,6 +32,11 @@ Firebase Core includes FIRApp and FIROptions which provide central configuration
   s.dependency 'GoogleUtilities/Logger', '~> 6.0'
   s.dependency 'FirebaseCoreDiagnosticsInterop', '~> 0.1'
 
+  # This dependency can be omitted by setting the environment variable DISABLE_FIREBASE_DIAGNOSTICS=1
+  unless ENV['DISABLE_FIREBASE_DIAGNOSTICS'] && ENV['DISABLE_FIREBASE_DIAGNOSTICS'] == '1' then
```

Comment:
Will this work after the podspec is published as json?

Reply:
Actually, I don't think so. `pod ipc spec` produces this:

```
{
  "name": "FirebaseCoreDiagnostics",
  "version": "0.1.0",
  "summary": "Firebase Core Diagnostics",
  "description": "Firebase Core Diagnostics collects diagnostic data to help improve and provide Firebase services.\nThis SDK is integrated using a 'soft-link' mechanism and can be omitted by having the environment\nvariable FIREBASE_DIAGNOSTICS set to 0.",
  "homepage": "https://firebase.google.com",
  "license": {
    "type": "Apache",
    "file": "LICENSE"
  },
  "authors": "Google, Inc.",
  "source": {
    "git": "https://github.com/firebase/firebase-ios-sdk.git",
    "tag": "CoreDiagnostics-0.1.0"
  },
  "social_media_url": "https://twitter.com/Firebase",
  "platforms": {
    "ios": "8.0",
    "osx": "10.11",
    "tvos": "10.0"
  },
  "cocoapods_version": ">= 1.4.0",
  "static_framework": true,
  "prefix_header_file": false,
  "pod_target_xcconfig": {
    "GCC_C_LANGUAGE_STANDARD": "c99",
    "GCC_TREAT_WARNINGS_AS_ERRORS": "YES",
    "CLANG_UNDEFINED_BEHAVIOR_SANITIZER_NULLABILITY": "YES",
    "GCC_PREPROCESSOR_DEFINITIONS": "PB_FIELD_32BIT=1 PB_NO_PACKED_STRUCTS=1 PB_ENABLE_MALLOC=1 FIRCoreDiagnostics_VERSION=0.1.0",
    "HEADER_SEARCH_PATHS": "\"${PODS_TARGET_SRCROOT}/Firebase/CoreDiagnostics/\""
  },
  "source_files": "Firebase/CoreDiagnostics/FIRCDLibrary/**/*.[cmh]",
  "private_header_files": "Firebase/CoreDiagnostics/FIRCDLibrary/Private/*.h",
  "frameworks": "Foundation",
  "dependencies": {
    "FirebaseCoreDiagnosticsInterop": [
      "~> 0.1"
    ],
    "GoogleDataTransportCCTSupport": [
      "~> 0.1"
    ],
    "GoogleUtilities/Environment": [
      "~> 6.0"
    ],
    "GoogleUtilities/Logger": [
      "~> 6.0"
    ]
  }
}
```

Reply:
@mikehaney24 @paulb777  Just wonder how it suppose to work. Should the user set `DISABLE_FIREBASE_DIAGNOSTICS = 1` before `pod update`?

Reply:
So it seems the ENV stuff should be deleted since it doesn't work.

There won't be a way to exclude the Data Logger from CocoaPods builds unless we want to make it opt in.

Reply:
As an option (not the best I guess), we can wrap the  `#if DISABLE_FIREBASE_DIAGNOSTICS = 1` FCD code in `FirebaseCoreDiagnostics` pod itself. Then the pod will be still installed but there will not be actually code to compile in it if the user defines the flag.

Reply:
Removed the env stuff. Removal story is same as before--using non-Cocoapods integration is the way to go. A #define won't work because Carthage and zip file are binary distributions.